File: app/src/main/java/com/stevenfrew/beatprompter/midi/ProgramChangeMessage.java
package com.stevenfrew.beatprompter.midi;

class ProgramChangeMessage extends OutgoingMessage {

    private static byte PROGRAM_CHANGE_MESSAGE_BYTE=(byte)0xc0;

    ProgramChangeMessage(int value, int channel)
    {
        super(mergeMessageByteWithChannel(PROGRAM_CHANGE_MESSAGE_BYTE,(byte)channel),(byte)value);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/comm/midi/message/ProgramChangeMessage.kt
package com.stevenfrew.beatprompter.comm.midi.message

internal class ProgramChangeMessage(
	value: Int,
	channel: Int
) : MidiMessage(
	mergeMessageByteWithChannel(MIDI_PROGRAM_CHANGE_BYTE, channel.toByte()),
	value.toByte()
)